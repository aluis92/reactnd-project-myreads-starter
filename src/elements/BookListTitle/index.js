import React from 'react';
import { string } from 'prop-types';

import {
    Container,
    Title,
} from './styles';

function BookListTitle({
    title,
}) {
    return (
        <Container className="list-books-title">
            <Title>{title}</Title>
        </Container>
    );
}

BookListTitle.defaultProps = {
    title: '',
};

BookListTitle.propTypes = {
    title: string.isRequired,
};

export default BookListTitle;
