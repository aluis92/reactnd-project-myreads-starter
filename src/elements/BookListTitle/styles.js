import styled from 'styled-components';

export const Container = styled.header`
    padding: 10px 0;
    background: #2e7c31;
    text-align: center;
`;

export const Title = styled.h1`
    font-weight: 400;
    margin: 0;
    color: white;
`;
