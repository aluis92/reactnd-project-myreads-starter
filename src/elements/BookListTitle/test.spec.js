import React from 'react';
import {
    shallow,
} from 'enzyme';

import BookListTitle from './index';

describe('BookListTitle component', () => {
    it('should render if it has a child', () => {
        expect(shallow(
            <BookListTitle>something</BookListTitle>,
        )).toMatchSnapshot();
    });
});
