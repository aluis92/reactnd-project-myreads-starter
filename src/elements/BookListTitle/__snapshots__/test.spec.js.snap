// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookListTitle component should render if it has a child 1`] = `
<styled.header>
  <styled.h1>
    something
  </styled.h1>
</styled.header>
`;
